uaclient/api/u/pro/config/v1.py

```diff
@@ -0,0 +1,146 @@
+    IntDataValue,
+    StringDataValue,
+)
+from uaclient.files import user_config_file
+
+LOG = logging.getLogger(util.replace_top_level_logger_name(__name__))
+
+
+class ConfigInfo(DataObject, AdditionalInfo):
+    fields = [
+        Field("http_proxy", StringDataValue, required=False, doc="HTTP proxy"),
+        Field(
+            "https_proxy", StringDataValue, required=False, doc="HTTPS proxy"
+        ),
+        Field(
+            "ua_apt_http_proxy",
+            StringDataValue,
+            required=False,
+            doc="UA APT HTTP proxy",
+        ),
+        Field(
+            "ua_apt_https_proxy",
+            StringDataValue,
+            required=False,
+            doc="UA APT HTTPS proxy",
+        ),
+        Field(
+            "global_apt_http_proxy",
+            StringDataValue,
+            required=False,
+            doc="Global APT HTTP proxy",
+        ),
+        Field(
+            "global_apt_https_proxy",
+            StringDataValue,
+            required=False,
+            doc="Global APT HTTPS proxy",
+        ),
+        Field("apt_news", BoolDataValue, required=False, doc="APT news"),
+        Field(
+            "apt_news_url", StringDataValue, required=False, doc="APT news URL"
+        ),
+        Field(
+            "metering_timer",
+            IntDataValue,
+            required=False,
+            doc="Metering timer",
+        ),
+        Field(
+            "update_messaging_timer",
+            IntDataValue,
+            required=False,
+            doc="Update messaging timer",
+        ),
+    ]
+
+    def __init__(
+        self,
+        *,
+        http_proxy: Optional[str] = None,
+        https_proxy: Optional[str] = None,
+        ua_apt_http_proxy: Optional[str] = None,
+        ua_apt_https_proxy: Optional[str] = None,
+        global_apt_http_proxy: Optional[str] = None,
+        global_apt_https_proxy: Optional[str] = None,
+        update_messaging_timer: Optional[int] = None,
+        metering_timer: Optional[int] = None,
+        apt_news: Optional[bool] = None,
+        apt_news_url: Optional[str] = None
+    ):
+        self.http_proxy = http_proxy
+        self.https_proxy = https_proxy
+        self.ua_apt_http_proxy = ua_apt_http_proxy
+        self.ua_apt_https_proxy = ua_apt_https_proxy
+        self.global_apt_http_proxy = global_apt_http_proxy
+        self.global_apt_https_proxy = global_apt_https_proxy
+        self.update_messaging_timer = update_messaging_timer
+        self.metering_timer = metering_timer
+        self.apt_news = apt_news
+        self.apt_news_url = apt_news_url
+
+
+def config() -> ConfigInfo:
+    return _config(UAConfig())
+
+
+def _config(cfg: UAConfig) -> ConfigInfo:
+    """This endpoint returns the current user configuration"""
+    pro_config = user_config_file.user_config.read().to_dict()
+    for key in UA_CONFIGURABLE_KEYS:
+        if hasattr(cfg, key) and pro_config[key] is None:
+            pro_config[key] = getattr(cfg, key)
```

Comment:
I think this can actually be
```suggestion
    pro_config = {}
    for key in UA_CONFIGURABLE_KEYS:
        if hasattr(cfg, key):
            pro_config[key] = getattr(cfg, key)
```

Reply:
Updated @orndorffgrant